src/core/server/integration_tests/saved_objects/migrations/check_registered_types.test.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { ISavedObjectTypeRegistry } from '@kbn/core-saved-objects-server';
+import { getMigrationHash } from '@kbn/core-test-helpers-so-type-serializer';
+import { Root } from '../../../root';
+import * as kbnTestServer from '../../../../test_helpers/kbn_server';
+
+describe('checking migration metadata changes on all registered SO types', () => {
+  let esServer: kbnTestServer.TestElasticsearchUtils;
+  let root: Root;
+  let typeRegistry: ISavedObjectTypeRegistry;
+
+  beforeAll(async () => {
+    const { startES } = kbnTestServer.createTestServers({
+      adjustTimeout: (t: number) => jest.setTimeout(t),
+    });
+
+    esServer = await startES();
+    root = kbnTestServer.createRootWithCorePlugins({}, { oss: false });
+    await root.preboot();
+    await root.setup();
+    const coreStart = await root.start();
+    typeRegistry = coreStart.savedObjects.getTypeRegistry();
+  });
+
+  afterAll(async () => {
+    if (root) {
+      await root.shutdown();
+    }
+    if (esServer) {
+      await esServer.stop();
+    }
+  });
+
+  // This test is meant to fail when any change is made in registered types that could potentially impact the SO migration.
+  // Just update the snapshot by running this test file via jest_integration with `-u` and push the update.
+  // The intent is to trigger a code review from the Core team to review the SO type changes.
+  it('detecting migration related changes in registered types', () => {
+    const allTypes = typeRegistry.getAllTypes();
+
+    const hashMap = allTypes.reduce((map, type) => {
+      map[type.name] = getMigrationHash(type);
+      return map;
+    }, {} as Record<string, string>);
+
+    expect(hashMap).toMatchInlineSnapshot(`
+      Object {
+        "action": "7858e6d5a9f231bf23f6f2e57328eb0095b26735",
+        "action_task_params": "bbd38cbfd74bf6713586fe078e3fa92db2234299",
+        "alert": "48461f3375d9ba22882ea23a318b62a5b0921a9b",
+        "api_key_pending_invalidation": "9b4bc1235337da9a87ef05a1d1f4858b2a3b77c6",
+        "apm-indices": "ceb0870f3a74e2ffc3a1cd3a3c73af76baca0999",
+        "apm-server-schema": "2bfd2998d3873872e1366458ce553def85418f91",
```

Comment:
So, we can't use dynamic snapshot easily to have one snapshot per type, at least not if we want to keep the snapshot inline (and even with non-inline, looping has its flaws)

So I had two options here:
1. use a single snapshot for all types
2. not use snapshot at all, and have a manually maintained map in the test field.

`2.` would have helped to have a more explicit error message (we could throw something like the comment at the top of the test), however the main upside I see with `1.` is that we can eventually, as a follow-up, run this test with `-u` as a pre-run CI task (like it's already done for linting or codeowners file update). If we achieve this, it would avoid the PR author to do anything manually. CI would simply run the test with `-u`, update the snapshot, push the change, which would trigger our review. 

This is for this reason (even if not implemented in current PR, we will have a follow-up for that) that I went with the 'big single snapshot' approach.